python/lsst/ap/association/assoc_afw_utils.py

```diff
@@ -0,0 +1,279 @@
+    schema.addField("psFluxErr", type='D')
+    schema.addField("filterName", type='String', size=10)
+    schema.addField("filterId", type='L')
+    return schema
+
+
+def ccdVisitSchema():
+    """Define the schema for the CcdVisit table.
+
+    Return
+    ------
+    ccdVisitNames: `collections.OrderedDict`
+       Names of columns in the ccdVisit table.
+    """
+    return oDict([("ccdVisitId", "INTEGER PRIMARY KEY"),
+                  ("ccdNum", "INTEGER"),
+                  ("filterName", "TEXT"),
+                  ("filterId", "INTEGER"),
+                  ("ra", "REAL"),
+                  ("decl", "REAL"),
+                  ("expTime", "REAL"),
+                  ("expMidptMJD", "REAL"),
+                  ("fluxZeroPoint", "REAL"),
+                  ("fluxZeroPointErr", "REAL")])
+
+
+def get_ccd_visit_info_from_exposure(exposure):
+    """
+    Extract info on the ccd and visit from the exposure.
+
+    Paramters
+    ---------
+    exposure : `lsst.afw.image.Exposure`
+        Exposure to store information from.
+
+    Returns
+    -------
+    values : `dict` of ``values``
+        List values representing info taken from the exposure.
+    """
+    visit_info = exposure.getInfo().getVisitInfo()
+    sphPoint = exposure.getWcs().getSkyOrigin()
+    flux0, flux0_err = exposure.getCalib().getFluxMag0()
+    filter_obj = exposure.getFilter()
+    # Values list is:
+    # [CcdVisitId ``int``,
+    #  ccdNum ``int``,
+    #  filterName ``str``,
+    #  RA WCS center ``degrees``,
+    #  DEC WCS center ``degrees``,
+    #  exposure time ``seconds``,
+    #  dateTimeMJD ``seconds``,
+    #  flux zero point ``counts``,
+    #  flux zero point error ``counts``]
+    values = {'ccdVisitId': visit_info.getExposureId(),
+              'ccdNum': exposure.getDetector().getId(),
+              'filterName': filter_obj.getName(),
+              'filterId': filter_obj.getId(),
+              'ra': sphPoint.getRa().asDegrees(),
+              'decl': sphPoint.getDec().asDegrees(),
+              'expTime': visit_info.getExposureTime(),
+              'expMidptMJD': visit_info.getDate().nsecs() * 10 ** -9,
+              'fluxZeroPoint': flux0,
+              'fluxZeroPointErr': flux0_err}
+    return values
+
+
+def add_dia_source_aliases_to_catalog(source_catalog):
+    """Create the needed aliases on fluxes and other columns if the input
+    catalog does not already have them.
+
+    Parameters
+    ----------
+    source_catalog : `lsst.afw.table.SourceCatalog`
+        Input source catalog to add alias columns to.
+    """
+    if not source_catalog.getSchema().contains(make_minimal_dia_source_schema()):
+        # Create aliases to appropriate flux fields if they exist.
+        schema_names = source_catalog.getSchema().getNames()
+        if 'base_PsfFlux_flux' in schema_names and \
+           'base_PsfFlux_fluxSigma' in schema_names and \
+           'psFlux' not in schema_names and \
+           'psFluxErr' not in schema_names:
```

Comment:
I'm confused about the purpose of this `if` statement. When would the latter two fields not be in the schema, but the former two would be?

Reply:
This is just to catch cases where someone may want to store a catalog that already has these aliases made or if I want to do a test case where I input a schema has psFlux etc.

Reply:
Ah the joys of aliasing.